# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/ClientWebGLContext.h
# Commit: b15d91e64a25
# Full Hash: b15d91e64a2589d47ad023d61dbd45135669eb60
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-11 04:04:33
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - NON_PREMULT TextureClient iff NON_PREMULT CanvasClient. r=handyman
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D78800
# ==============================================================================

diff -r 71ad2ed8e9ba -r b15d91e64a25 dom/canvas/ClientWebGLContext.h
--- a/dom/canvas/ClientWebGLContext.h	Wed Jun 10 22:21:07 2020 +0000
+++ b/dom/canvas/ClientWebGLContext.h	Wed Jun 10 22:21:09 2020 +0000
@@ -723,8 +723,6 @@
   bool mIsCanvasDirty = false;
   uvec2 mRequestedSize = {};
 
-  RefPtr<gfx::DataSourceSurface> mFrontBufferSnapshot;
-
  public:
   explicit ClientWebGLContext(bool webgl2);
 
@@ -997,7 +995,8 @@
   void Present(WebGLFramebufferJS*, layers::TextureType);
   Maybe<layers::SurfaceDescriptor> GetFrontBuffer(WebGLFramebufferJS*,
                                                   layers::TextureType);
-  RefPtr<gfx::SourceSurface> GetFrontBufferSnapshot() override;
+  RefPtr<gfx::SourceSurface> GetFrontBufferSnapshot(
+      bool requireAlphaPremult = true) override;
 
  private:
   RefPtr<gfx::DataSourceSurface> BackBufferSnapshot();